application.go

```diff
@@ -431,12 +429,14 @@ func (r *marathonClient) WaitOnApplication(name string, timeout time.Duration) e
 			flick.SwitchOn()
 		}()
 		for !flick.IsSwitched() {
-			if found, err := r.HasApplication(name); err != nil {
-				continue
-			} else if found {
-				if app, err := r.Application(name); err == nil && app.AllTaskRunning() {
-					return nil
+			if err := r.HasApplication(name); err != nil {
+				if err != ErrDoesNotExist {
+					return err
```

Comment:
The original code is a bit hard to understand, but I think you may have inadvertently deviated from the existing logic along your refactoring:

- When the application does not exist in the old implementation (`found` == `false` and `err` == `nil`), we sleep 500 ms. In the new implementation where _not found_ is exposed via the `ErrDoesNotExist` error, we just continue.
- When `HasApplication` returns an error in the old implementation (except for `ErrDoesNotExist` because that's mapped to a `false` result), we continue. In the new implementation, we return the error to the caller for anything other than `ErrDoesNotExist`.